### DISCUSSION TITLE
Managing Javascript includes on a page

### DISCUSSION THREAD

**Discussion by stljeff1** [GrapesJS/grapesjs#3801] [Q&A]
Does anybody have ideas on how to manage script tags on a page? 

Say for instance I create a block that requires jQuery, and maybe another block that requires jQuery and a carousel plugin. How can I maintain the proper order of the script tags as I add blocks to a page? How could I inject a library script (like jQuery) that needs to be the first script tag of the page (which would come after all of the HTML on the page)?

**Comment by ronaldohoch** [comment]
Hello!

I have made a plugin to manage it, i think this two commands can help you.:

```javascript
    Commands.add("dina:get-styles",(editor,sender,opts={}) => {
        var head = editor.Canvas.getDocument().head;
        var link = head.querySelectorAll("link");
        var files = []
        
        link.forEach(el=>{
            files.push(el.outerHTML);
        });
        return files.join("");
    });

    Commands.add("dina:get-scripts",(editor,sender,opts={}) => {
        var head = editor.Canvas.getDocument().head;
        var script = head.querySelectorAll("script");
        var files = []
        
        script.forEach(el=>{
            files.push(el.outerHTML);
        });
        return files.join("");
    });
```

![image](https://user-images.githubusercontent.com/2287371/134405970-b9b7474a-a040-481e-a2c7-6cba5a15ec37.png)
